sklearn/model_selection/tests/test_classification_threshold.py

```diff
@@ -0,0 +1,938 @@
+        else:  # "max_tpr_at_tnr_constraint" or "max_recall_at_precision_constraint"
+            assert 0 <= model.best_threshold_ <= 0.5
+    else:  # "decision_function"
+        # we expect the decision function to be centered in 0.0 and to be larger than
+        # -1 and 1.
+        if objective_metric in (
+            "max_tnr_at_tpr_constraint",
+            "max_precision_at_recall_constraint",
+        ):
+            assert 0 < model.best_threshold_ < 20
+        else:  # "max_tpr_at_tnr_constraint" or "max_recall_at_precision_constraint"
+            assert -20 < model.best_threshold_ < 0
+
+
+@pytest.mark.usefixtures("enable_slep006")
+def test_tuned_threshold_classifier_cv_zeros_sample_weights_equivalence():
+    """Check that passing removing some sample from the dataset `X` is
+    equivalent to passing a `sample_weight` with a factor 0."""
+    X, y = load_iris(return_X_y=True)
+    # Scale the data to avoid any convergence issue
+    X = StandardScaler().fit_transform(X)
+    # Only use 2 classes and select samples such that 2-fold cross-validation
+    # split will lead to an equivalence with a `sample_weight` of 0
+    X = np.vstack((X[:40], X[50:90]))
+    y = np.hstack((y[:40], y[50:90]))
+    sample_weight = np.zeros_like(y)
+    sample_weight[::2] = 1
+
+    estimator = LogisticRegression().set_fit_request(sample_weight=True)
+    model_without_weights = TunedThresholdClassifierCV(estimator, cv=2)
+    model_with_weights = clone(model_without_weights)
+
+    model_with_weights.fit(X, y, sample_weight=sample_weight)
+    model_without_weights.fit(X[::2], y[::2])
+
+    assert_allclose(
+        model_with_weights.estimator_.coef_, model_without_weights.estimator_.coef_
+    )
+
+    y_pred_with_weights = model_with_weights.predict_proba(X)
+    y_pred_without_weights = model_without_weights.predict_proba(X)
+    assert_allclose(y_pred_with_weights, y_pred_without_weights)
+
+
+def test_tuned_threshold_classifier_error_constant_learner():
+    """Check that we raise an error message when providing an estimator that predicts
+    only a single class."""
+    X, y = make_classification(random_state=0)
+    estimator = DummyClassifier(strategy="constant", constant=1)
+    err_msg = "The provided estimator makes constant predictions."
+    with pytest.raises(ValueError, match=err_msg):
+        TunedThresholdClassifierCV(estimator).fit(X, y)
+
+
+@pytest.mark.parametrize(
+    "objective_metric",
+    ["max_precision_at_recall_constraint", "max_recall_at_precision_constraint"],
+)
+@pytest.mark.parametrize("pos_label", [0, 1])
+def test_tuned_threshold_classifier_pos_label_precision_recall(
+    objective_metric, pos_label
+):
+    """Check that `pos_label` is dispatched correctly by checking the precision and
+    recall score found during the optimization and the one found at `predict` time."""
+    X, y = make_classification(n_samples=5_000, weights=[0.6, 0.4], random_state=42)
```

Comment:
They look redundant but we check each strategy (roc curve, pr curve, and metric) and make sure that we get the proper attributes after fit. I think this is better to have those tests since they covered different branch.